src/mudlet.cpp

```diff
@@ -2335,37 +2331,53 @@ void mudlet::slot_replaySpeedDown()
     replaySpeedDisplay->show();
 }
 
+/* loop through and stop all sounds */
 void mudlet::stopSounds()
 {
-    mpMusicBox1->stop();
-    mpMusicBox2->stop();
-    mpMusicBox3->stop();
-    mpMusicBox4->stop();
+    QListIterator<QMediaPlayer *> itMusicBox( MusicBoxList );
+    
+    while( itMusicBox.hasNext() ) {
+        itMusicBox.next()->stop();
+    }
 }
 
-void mudlet::playSound( QString s )
+void mudlet::playSound( QString s, int soundVolume )
 {
-    if( mpMusicBox1->state() != QMediaPlayer::PlayingState )
-    {
-        mpMusicBox1->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox1->play();
-    }
-    else if( mpMusicBox2->state() != QMediaPlayer::PlayingState )
-    {
-        mpMusicBox2->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox2->play();
-    }
-    else if( mpMusicBox3->state() != QMediaPlayer::PlayingState )
-    {
-        mpMusicBox3->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox3->play();
+    QListIterator<QMediaPlayer *> itMusicBox( MusicBoxList );
+    QMediaPlayer * pPlayer=0;
+    
+    /* find first available inactive QMediaPlayer */
+    while( itMusicBox.hasNext() ) {
+        QMediaPlayer * pTestPlayer=itMusicBox.next();
+        
+        if ( pTestPlayer->state() != QMediaPlayer::PlayingState && pTestPlayer->mediaStatus() != QMediaPlayer::LoadingMedia ) {
+            pPlayer=pTestPlayer;
+            break;
+        }
     }
-    else
-    {
-        mpMusicBox4->stop();
-        mpMusicBox4->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox4->play();
+
+    /* no available QMediaPlayer, create a new one */
+    if (!pPlayer) {
+        pPlayer = new QMediaPlayer(this);
+
+        if (!pPlayer) {
+            /* It (should) be impossible to ever reach this */
+            Host * pH = getActiveHost();
+            
+            if (pH) {
+                pH->postMessage( "\n[  ERROR  ]  - Unable to create new QMediaPlayer object\n" );    
```

Comment:
:bug: Also, a leading new-line will __break__ the code that detects the _type_ of message that was posted and applies the _colour coding_ to it...